File: Src/java/elm-jackson/src/main/java/org/cqframework/cql/elm/serializing/jackson/ElmLibraryWriterProvider.java
package org.cqframework.cql.elm.serializing.jackson;

import org.cqframework.cql.elm.serializing.ElmLibraryWriter;

public class ElmLibraryWriterProvider implements org.cqframework.cql.elm.serializing.ElmLibraryWriterProvider {
    @Override
    public ElmLibraryWriter create(String contentType) {
        if (contentType == null) {
            contentType = "application/elm+json";
        }
        switch (contentType) {
            case "application/elm+xml": return new ElmXmlLibraryWriter();
            case "application/elm+json":
            default: return new ElmJsonLibraryWriter();
        }
    }
}

File: Src/java/elm-xmlutil/src/main/java/org/cqframework/cql/elm/serializing/xmlutil/ElmLibraryWriterProvider.kt
package org.cqframework.cql.elm.serializing.xmlutil

import org.cqframework.cql.elm.serializing.ElmLibraryWriter
import org.cqframework.cql.elm.serializing.ElmLibraryWriterProvider

class ElmLibraryWriterProvider : ElmLibraryWriterProvider {
    override fun create(contentType: String): ElmLibraryWriter {
        var contentType: String? = contentType
        if (contentType == null) {
            contentType = "application/elm+json"
        }
        return when (contentType) {
            "application/elm+xml" -> ElmXmlLibraryWriter()
            "application/elm+json" -> ElmJsonLibraryWriter()
            else -> ElmXmlLibraryWriter()
        }
    }
}
